parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeospatialUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.column.statistics.geometry;
+
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.CoordinateSequenceFilter;
+import org.locationtech.jts.geom.Geometry;
+
+class GeospatialUtils {
+
+  public static void normalizeLongitude(Geometry geometry) {
+    if (geometry == null || geometry.isEmpty()) {
+      return;
+    }
+
+    geometry.apply(new CoordinateSequenceFilter() {
+      @Override
+      public void filter(CoordinateSequence seq, int i) {
+        double x = seq.getX(i);
+        // Normalize the longitude to be within -180 to 180 range
+        while (x > 180) x -= 360;
+        while (x < -180) x += 360;
```

Comment:
Another reason why we don't want to update geospatial statistics for plain binary column is that, when the binary column happens to be a valid WKB, and the x coordinate parsed from that WKB is extremely large, we may end up stuck in these loops.

Also, we can replace these loops with simple arithmetics. Here is an example:

```java
private double normalizeLongitude(double lon) {
  if (lon >= -180.0 && lon <= 180.0) {
    return lon;
  } else if (lon > 180) {
    return (lon + 180.0) % 360.0 - 180.0;
  } else {
    return 180.0 - (180.0 - lon) % 360.0;
  }
}
```